Source/CSharpEssentials/Extensions.cs

```diff
@@ -168,9 +170,64 @@ private static bool IsGeneratedCode(string filePath)
             return false;
         }
 
-        public static bool IsGeneratedCode(this SyntaxTree tree)
+        public static bool IsGeneratedCode(this SyntaxNode node)
         {
-            return IsGeneratedCode(tree.FilePath);
+            return IsInGeneratedCodeContext(node)
+                || IsGeneratedCode(node.SyntaxTree.FilePath);
+        }
+
+        private static bool IsInGeneratedCodeContext(this SyntaxNode node)
+        {
+            do
+            {
+                bool inGeneratedCodeContext = false;
+                switch (node.Kind())
```

Comment:
Yup. You missed at least EventFieldDeclarations. Getting this right is hard. You might just try simple 'as' checks to ```BaseMethodDeclarationSyntax```, ```BasePropertyDeclarationSyntax```, ```BaseTypeDeclarationSyntax```, etc. AttributeLists should be declared on those base types.